litestar/contrib/pydantic/utils.py

```diff
@@ -52,13 +53,15 @@ def is_pydantic_model_class(
     Returns:
         A typeguard determining whether the type is :data:`BaseModel pydantic.BaseModel>`.
     """
-    if pydantic_v1 is Empty:  # type: ignore[comparison-overlap] # pragma: no cover
-        return False
+    tests: list[bool] = []
 
-    if pydantic_v2 is Empty:  # type: ignore[comparison-overlap] # pragma: no cover
-        return is_class_and_subclass(annotation, pydantic_v1.BaseModel)
+    if pydantic_v1 is not Empty:  # pragma: no cover
+        tests.append(is_class_and_subclass(annotation, pydantic_v1.BaseModel))
 
-    return is_class_and_subclass(annotation, (pydantic_v1.BaseModel, pydantic_v2.BaseModel))
+    if pydantic_v2 is not Empty:  # pragma: no cover
+        tests.append(is_class_and_subclass(annotation, pydantic_v2.BaseModel))
+
+    return any(tests)
```

Comment:
there shouldn't be any functional change, just pyright was a lot happier with this pattern.